x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/multivalue/MvPSeriesWeightedSum.java

```diff
@@ -130,10 +135,13 @@ protected NodeInfo<? extends Expression> info() {
 
     @Override
     public DataType dataType() {
+        if (p.dataType() == NULL) {
```

Comment:
Do we really need this? Yes, it _will_ return a NULL. But it's because of a planner rule, not because this functions can return nulls in any way (If that makes sense?).

This is not how other functions work, so I feel like we're overcomplicating things here